packages/react-storage/src/components/StorageBrowser/views/LocationsView/__tests__/LocationsView.spec.tsx

```diff
@@ -290,7 +318,80 @@ describe('LocationsListView', () => {
     expect(handleListLocations).toHaveBeenCalledTimes(1);
     expect(updatedHandleListLocations).toHaveBeenCalledTimes(1);
     expect(updatedHandleListLocations).toHaveBeenCalledWith({
-      options: { exclude: 'WRITE', pageSize: 100, refresh: true },
+      options: {
+        exclude: 'WRITE',
+        pageSize: EXPECTED_PAGE_SIZE,
+        refresh: true,
+      },
+    });
+  });
+
+  it('can paginate forward and back', async () => {
+    useLocationsDataSpy.mockReturnValue(resolvedState);
+    render(
+      <Provider>
+        <LocationsView />
+      </Provider>
+    );
+
+    // table renders
+    const table = screen.queryByRole('table');
+    expect(table).toBeInTheDocument();
+
+    // pagination enabled
+    const nextPage = await screen.findByLabelText('Go to next page');
+    expect(nextPage).not.toBeDisabled();
+
+    // first page data matches input
+    expect(screen.queryByLabelText('Page 1')).toBeInTheDocument();
+    expect(screen.queryByText('test-bucket-0')).toBeInTheDocument();
+    expect(screen.queryByText('next-page-0')).not.toBeInTheDocument();
+
+    useLocationsDataSpy.mockReturnValue(nextPageState);
+
+    // go forward
+    await act(async () => {
```

Comment:
we shouldn't have to add `act` here -- but we have [this issue](https://github.com/testing-library/user-event/discussions/906#discussioncomment-2522468) in the repo